server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -40,8 +40,7 @@ def update
       # controller (after it has validated them), never by directly updating
       # the deployment object.
 
-      # commented out :openstack_undercloud_password since mock UI application needs to set this value
-      # params[:deployment].delete :openstack_undercloud_password
+      params[:deployment].delete :openstack_undercloud_password
```

Comment:
@isratrade Not sure if this is the right fix, since I wasn't clear on why a mock UI application would need to update this on the deployment object. It was keeping @jmontleon from deploying as the put was setting openstack_undercloud_password to nil, though.  There are other ways of fixing it (ie reloading deployment after the value is set) but we can discuss the issue when you get back into work.